qiskit/pulse/channels/output_channel.py

```diff
@@ -9,26 +9,75 @@
 Channels support signal output.
 """
 from abc import abstractmethod
+from typing import Tuple
 
+from qiskit.pulse.exceptions import PulseError
 from .pulse_channel import Channel
 
 
+class LoRange:
+    """Range of LO frequency."""
+
+    def __init__(self, lower_bound: float, upper_bound: float):
+        self._lb = lower_bound
+        self._ub = upper_bound
+
+    def includes(self, lo_freq: float) -> bool:
+        """Return if this range includes `lo_freq` or not.
+
+        Args:
+            lo_freq: LO frequency to be checked
+
+        Returns:
+            bool: True if lo_freq is included in this range, otherwise False
+        """
+        if self._lb <= lo_freq <= self._ub:
+            return True
+        return False
+
+    @property
+    def lower_bound(self) -> float:
+        """Lower bound of this LO range"""
+        return self._lb
+
+    @property
+    def upper_bound(self) -> float:
+        """Upper bound of this LO range"""
+        return self._ub
+
+    def __repr__(self):
+        return "%s(%f, %f)" % (self.__class__.__name__, self._lb, self._ub)
+
+
 class OutputChannel(Channel):
     """Output Channel."""
 
     @abstractmethod
-    def __init__(self, index: int = None, lo_frequency: float = None):
+    def __init__(self,
+                 index: int,
+                 lo_frequency: float = None,
+                 lo_freq_range: Tuple[float, float] = (0, float("inf"))):
         super().__init__(index)
         self._lo_frequency = lo_frequency
 
+        if (not isinstance(lo_freq_range, tuple)) or len(lo_freq_range) != 2:
+            raise PulseError("Invalid form of lo_freq_range is specified.")
+        self._lo_freq_range = LoRange(*lo_freq_range)
+
+        if self._lo_frequency:
+            if not self._lo_freq_range.includes(self._lo_frequency):
+                raise PulseError("lo_frequency %f must be within lo_freq_range %s" %
+                                 (self._lo_frequency, self._lo_freq_range))
+
     @property
     def lo_frequency(self) -> float:
-        """Get the frequency of local oscillator of this channel."""
+        """Get the default frequency of local oscillator of this channel."""
         return self._lo_frequency
 
-    def set_lo_frequency(self, lo_frequency: float):
-        """Set the frequency of local oscillator of this channel."""
-        self._lo_frequency = lo_frequency
+    @property
+    def lo_freq_range(self) -> LoRange:
+        """Get the feasible range of LO frequency."""
+        return self._lo_freq_range
 
     def __eq__(self, other):
         """Two output channels are the same if they are of the same type, and
```

Comment:
Should the range be enforced for equality? Should it not just be `index` and channel type?

Reply:
You're right. I've just overlooked.